cmake/scripts/common/PrepareEnv.cmake

```diff
@@ -7,6 +7,18 @@ if(NOT DEFINED KODI_MIRROR)
   set(KODI_MIRROR "http://mirrors.kodi.tv")
 endif()
 
+if(EXISTS ${CORE_SOURCE_DIR}/cmake/platform/${CORE_SYSTEM_NAME}/${CORE_PLATFORM_NAME_LC}.cmake)
+  include(${CORE_SOURCE_DIR}/cmake/platform/${CORE_SYSTEM_NAME}/${CORE_PLATFORM_NAME_LC}.cmake)
+  if(ENABLE_APP_AUTONAME)
+    set(APP_BINARY_SUFFIX "-${CORE_PLATFORM_NAME_LC}")
+  endif()
+else()
+  file(GLOB _platformnames RELATIVE ${CORE_SOURCE_DIR}/cmake/platform/${CORE_SYSTEM_NAME}/
+                                    ${CORE_SOURCE_DIR}/cmake/platform/${CORE_SYSTEM_NAME}/*.cmake)
+  string(REPLACE ".cmake" " " _platformnames ${_platformnames})
+  message(FATAL_ERROR "invalid CORE_PLATFORM_NAME: ${CORE_PLATFORM_NAME_LC}\nValid platforms: ${_platformnames}")
```

Comment:
With this c&p part from Platform.cmake I'm not sure, are still OS types defined in ./cmake/platform/ folder?